src/screens/user/MyJobScreen.tsx

```diff
@@ -0,0 +1,373 @@
+import * as Permissions from 'expo-permissions';
+import Constants from 'expo-constants';
+import { theme } from '../../core/theme';
+import gql from 'graphql-tag';
+import { useMutation, useQuery } from '@apollo/react-hooks';
+import { jobToGlobalState } from '../../store/user/actions';
+import { Avatar, Card, Title, Paragraph } from 'react-native-paper';
+
+type Props = {
+  navigation: Navigation;
+  user: { country: String; id: String; jobs: [] };
+  jobToGlobalState: (job: Object) => void;
+};
+
+//TODO: ADD type for images $images: [Upload!]! || single file not required: Upload
+const ADD_JOB_MUTATION = gql`
+  mutation AddJob(
+    $title: String!
+    $description: String!
+    $price: Int!
+    $country: String!
+    $city: String!
+    $postalCode: String!
+    $address: String!
+    $userId: String!
+    $jobCategoryId: String!
+  ) {
+    addJob(
+      title: $title
+      description: $description
+      price: $price
+      country: $country
+      city: $city
+      postalCode: $postalCode
+      address: $address
+      userId: $userId
+      jobCategoryId: $jobCategoryId
+    ) {
+      id
+      title
+      description
+      price
+      city
+      postalCode
+      address
+      userId
+      jobCategoryId
+    }
+  }
+`;
+
+const MyJobScreen = ({ navigation, user, jobToGlobalState }: Props) => {
+  const [uploadJob] = useMutation(ADD_JOB_MUTATION);
+  const [modalOpen, setModalOpen] = useState(false);
+  const [title, setTitle] = useState({ value: '', error: '' });
+  const [description, setDescription] = useState({ value: '', error: '' });
+  const [price, setPrice] = useState({ value: null, error: '' });
+  const [images, setImage] = useState({ uri: null, error: '' });
+  const [city, setCity] = useState({ value: null, error: '' });
+  const [postalCode, setPostalCode] = useState({ value: null, error: '' });
+  const [address, setAddress] = useState({ value: null, error: '' });
+  const [hasCameraPermission, setCameraPermission] = useState({
+    value: null,
+    error: '',
+  });
+
+  const validateInput = () => {
+    const titleError = titleValidator(title.value);
+    const descriptionError = descriptionValidator(description.value);
+    const priceError = priceValidator(price.value);
+    const cityError = cityValidator(city.value);
+    const postalCodeError = postalCodeValidator(postalCode.value);
+    const addressError = addressValidator(address.value);
+    if (titleError) {
+      setTitle({ ...title, error: titleError });
+    }
+    if (descriptionError) {
+      setDescription({ ...description, error: descriptionError });
+    }
+    if (priceError) {
+      setPrice({ ...price, error: priceError });
+    }
+    if (cityError) {
+      setCity({ ...city, error: cityError });
+    }
+    if (postalCodeError) {
+      setPostalCode({ ...postalCode, error: postalCodeError });
+    }
+    if (addressError) {
+      setAddress({ ...address, error: addressError });
+    }
```

Comment:
I think we can come up with something better for this. I think i got it, it might be a bit long but tell me if you follow:
```
// on src/core/utlis.js we add:
export const validate = {
  title: titleValidator,
  address: addressValidator,
  postalCode: postalCodeValidator,
  price: priceValidator,
  description: descriptionValidator,
  name: nameValidator,
  email: emailValidator,
};

// then in the component
import { validate } from '../../core/utils';

// add field name to state like this
const [title, setTitle] = useState({ name: 'title', value: '', error: '' });
const [city, setCity] = useState({ name: 'city', value: null, error: '' });
...

// define array with state and setters for each field
  const formData = [
    [title, setTitle],
    [description, setDescription],
    [price, setPrice],
    [city, setCity],
    [postalCode, setPostalCode],
    [address, setAddress],
  ];

// finally:

formData.forEach(field => {
      const error = validate[field[0].name](field[0].value);

       // if error we call the second element of the array "field" that is the state setter
      if (error) field[1]({ ...field[0], error });
    });
```
Haven't tried it but i think it should work, give it a shot!

Reply:
Beautiful. It works!